File: Assetraz/AssetrazAPI/AssetrazDataProvider/Entities/Vendor.cs
using System;
using System.Collections.Generic;

namespace AssetrazDataProvider.Entities
{
    public partial class Vendor
    {
        public Vendor()
        {
            PurchaseOrderHeaders = new HashSet<PurchaseOrderHeader>();
            RequestForQuoteDetails = new HashSet<RequestForQuoteDetail>();
            RequestForQuoteParticipants = new HashSet<RequestForQuoteParticipant>();
            VendorsCategories = new HashSet<VendorsCategory>();
        }

        public Guid VendorId { get; set; }
        public string VendorName { get; set; } = null!;
        public string VendorAddressLine1 { get; set; } = null!;
        public string? VendorAddressLine2 { get; set; }
        public string City { get; set; } = null!;
        public string State { get; set; } = null!;
        public string Country { get; set; } = null!;
        public string? Gstin { get; set; }
        public bool PreferredVendor { get; set; }
        public string ContactPerson { get; set; } = null!;
        public string ContactNumber { get; set; } = null!;
        public string EmailAddress { get; set; } = null!;
        public bool? Active { get; set; }
        public DateTime CreatedDate { get; set; }
        public string CreatedBy { get; set; } = null!;
        public DateTime UpdatedDate { get; set; }
        public string UpdatedBy { get; set; } = null!;

        public virtual ICollection<PurchaseOrderHeader> PurchaseOrderHeaders { get; set; }
        public virtual ICollection<RequestForQuoteDetail> RequestForQuoteDetails { get; set; }
        public virtual ICollection<RequestForQuoteParticipant> RequestForQuoteParticipants { get; set; }
        public virtual ICollection<VendorsCategory> VendorsCategories { get; set; }
    }
}

File: Assetraz/AssetrazUI/src/types/Vendor.ts
import React from "react";
export type Vendor = {
    vendorId: string;
    vendorName: string;
};

export interface IVendor {
    vendorId: string;
    vendorName: string;
    vendorAddressLine1: string;
    vendorAddressLine2?: string;
    city: string;
    state: string ;
    country: string;
    gstin?: string;
    preferredVendor: boolean;
    contactPerson: string;
    contactNumber: string ;
    emailAddress: string;
    createdDate: Date;
    createdBy: string;
    updatedDate: Date;
    updatedBy: string;
    active?: boolean;
}

export type VendorModalProps = {
    toggleShowModal: () => void;
    successMessage?: string;
    setSuccessMessage: (message: string | undefined) => void;
    getData: () => void;
    selectedVendor?: IVendor | undefined;
    isAddModal?: boolean;
    setIsAddModal?: React.Dispatch<React.SetStateAction<boolean>>;
};

export type AddVendorModalFooterProps = {
    closeModal: () => void;
    setLoading: React.Dispatch<React.SetStateAction<boolean>>;
    setErrorMessage: (message: string | undefined) => void;
    setSuccessMessage: (message: string | undefined) => void;
    getData: () => void;
    setIsAddModal?: React.Dispatch<React.SetStateAction<boolean>>;
    dataEdited?: boolean;
    isAddModal?: boolean;
};

export type AddVendorModalBodyProps = {
    isAddModal?: boolean;
    setDataEdited: React.Dispatch<React.SetStateAction<boolean>>;
};
